pkg/networking/ingress.go

```diff
@@ -0,0 +1,118 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/argoproj-labs/argocd-operator/pkg/argoutil"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IngressRequest objects contain all the required information to produce a ingress object in return
+type IngressRequest struct {
```

Comment:
@ciiay there are a few ingress specific fields that can be included here 
such as `spec.Rules`, `spec.IngressClassname` and `spec.TLS`
see https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/ingress.go#L123